### DISCUSSION TITLE
How do others model disks?

### DISCUSSION THREAD

**Discussion by mattboston** [netbox-community/netbox#9546] [General]
How do others keep track of disks in servers?  Most modern servers have disk sleds and are swappable.  But there are different sizes(physical) and types (sas, scsi, etc).

**Comment by candlerb** [comment]
You can list individual disks as [inventory items](https://docs.netbox.dev/en/stable/models/dcim/inventoryitem/) - including their type, serial number etc.

However, personally I find that is too hard work to keep up to date. Netbox is a source of truth for pushing out configurations, but I can't arrange that changing the serial number of a disk in Netbox causes the disk to be swapped automatically :-)

Instead, I use SMART and megaraid monitoring to generate [custom metrics](https://github.com/prometheus-community/node-exporter-textfile-collector-scripts) for prometheus node exporter, and that lets me see which disks are (or were) in any given server at any particular time.
